// table-oak/lib/FormatType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require("react");

exports.default = {
    TextField: function TextField(column, body) {
        var _this = this;

        return (0, _react.cloneElement)(column.type, {
            value: body[column.source] || "",
            name: column.source,
            onChange: function onChange(e) {
                body[column.source] = e.target.value;
                _this.setState(_this.state);
            },
            onClick: function onClick(e) {
                e.stopPropagation();
                e.preventDefault();
            }
        });
    },
    SelectField: function SelectField(column, body) {
        var _this2 = this;

        return (0, _react.cloneElement)(column.type, {
            value: body[column.source] || "",
            name: column.source,
            onChange: function onChange(event, index, value) {
                body[column.source] = value;
                _this2.setState(_this2.state);
            },
            onClick: function onClick(e) {
                e.stopPropagation();
                e.preventDefault();
            }
        });
    },
    AutoComplete: function AutoComplete(column, body) {
        var _this3 = this;

        return (0, _react.cloneElement)(column.type, {
            value: body[column.source] || "",
            searchText: body[column.source] || "",
            filter: function filter(searchText, key, data) {
                return data.inputCode.startsWith(searchText.toUpperCase());
            },
            name: column.source,
            onUpdateInput: function onUpdateInput(searchText) {
                body[column.source] = searchText;
                _this3.setState(_this3.state);
            },
            onClick: function onClick(e) {
                e.stopPropagation();
                e.preventDefault();
            }
        });
    }
}; /**
    * Created by jahv on 2017/5/6.
    */
