// common/component/loading/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _getPrototypeOf = require("babel-runtime/core-js/object/get-prototype-of");

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require("babel-runtime/helpers/classCallCheck");

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require("babel-runtime/helpers/createClass");

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require("babel-runtime/helpers/possibleConstructorReturn");

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require("babel-runtime/helpers/inherits");

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _temp2; /**
                     * Created by jahv on 2017/4/13.
                     */


var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _RefreshIndicator = require("material-oak/RefreshIndicator");

var _RefreshIndicator2 = _interopRequireDefault(_RefreshIndicator);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require("classnames");

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Loading = (_temp2 = _class = function (_Component) {
    (0, _inherits3.default)(Loading, _Component);

    function Loading() {
        var _ref;

        var _temp, _this, _ret;

        (0, _classCallCheck3.default)(this, Loading);

        for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
            args[_key] = arguments[_key];
        }

        return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = Loading.__proto__ || (0, _getPrototypeOf2.default)(Loading)).call.apply(_ref, [this].concat(args))), _this), _this.style = {
            container: {
                position: 'relative',
                width: "50px",
                margin: "0 auto"
            },
            refresh: {
                display: 'inline-block',
                position: 'relative'
            }
        }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
    }

    (0, _createClass3.default)(Loading, [{
        key: "render",
        value: function render() {
            var _style = this.style,
                container = _style.container,
                refresh = _style.refresh;
            var show = this.props.show;

            return _react2.default.createElement(
                "div",
                { className: (0, _classnames2.default)("loading_modal", { hide: !show }) },
                _react2.default.createElement(
                    "div",
                    { className: "", style: container },
                    _react2.default.createElement(_RefreshIndicator2.default, {
                        size: 50,
                        left: 0,
                        top: 20,
                        loadingColor: "#FF9800",
                        status: show ? "loading" : "hide",
                        style: refresh
                    })
                )
            );
        }
    }]);
    return Loading;
}(_react.Component), _class.propsTypes = {
    show: _propTypes2.default.boolean
}, _class.defaultProps = {
    show: false
}, _temp2);
exports.default = Loading;
module.exports = exports["default"];
