// table-oak/src/FormatType.js
/**
 * Created by jahv on 2017/5/6.
 */
import {cloneElement} from "react";

export default {
    TextField(column, body){
        return cloneElement(column.type, {
            value: body[column.source] || "",
            name: column.source,
            onChange: e => {
                body[column.source] = e.target.value;
                this.setState(this.state);
            },
            onClick(e){
                e.stopPropagation();
                e.preventDefault();
            }
        })
    },
    SelectField(column, body){
        return cloneElement(column.type, {
            value: body[column.source] || "",
            name: column.source,
            onChange: (event, index, value) => {
                body[column.source] = value;
                this.setState(this.state);
            },
            onClick(e){
                e.stopPropagation();
                e.preventDefault();
            }
        })
    },
    AutoComplete(column, body){
        return cloneElement(column.type, {
            value: body[column.source] || "",
            searchText: body[column.source] || "",
            filter: (searchText, key, data) => {
                return data.inputCode.startsWith(searchText.toUpperCase());
            },
            name: column.source,
            onUpdateInput: (searchText) => {
                body[column.source] = searchText;
                this.setState(this.state);
            },
            onClick(e){
                e.stopPropagation();
                e.preventDefault();
            }
        })
    }
}
