// common/component/frame/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _objectWithoutProperties2 = require("babel-runtime/helpers/objectWithoutProperties");

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require("babel-runtime/core-js/object/get-prototype-of");

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require("babel-runtime/helpers/classCallCheck");

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require("babel-runtime/helpers/createClass");

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require("babel-runtime/helpers/possibleConstructorReturn");

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require("babel-runtime/helpers/inherits");

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _BottomNavigation = require("material-oak/BottomNavigation");

var _FontIcon = require("material-oak/FontIcon");

var _FontIcon2 = _interopRequireDefault(_FontIcon);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Created by jahv on 2017/4/30.
 */
var Frame = function (_React$Component) {
    (0, _inherits3.default)(Frame, _React$Component);

    function Frame() {
        var _ref;

        var _temp, _this, _ret;

        (0, _classCallCheck3.default)(this, Frame);

        for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
            args[_key] = arguments[_key];
        }

        return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = Frame.__proto__ || (0, _getPrototypeOf2.default)(Frame)).call.apply(_ref, [this].concat(args))), _this), _this.propTypes = {
            close: _propTypes.boolean,
            max: _propTypes.boolean,
            min: _propTypes.boolean
        }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
    }

    (0, _createClass3.default)(Frame, [{
        key: "render",
        value: function render() {
            var _props = this.props,
                close = _props.close,
                max = _props.max,
                min = _props.min,
                other = (0, _objectWithoutProperties3.default)(_props, ["close", "max", "min"]);

            return _react2.default.createElement(
                _BottomNavigation.BottomNavigation,
                { className: "frame" },
                _react2.default.createElement(_BottomNavigation.BottomNavigationItem, {
                    className: "close",
                    icon: _react2.default.createElement(_FontIcon2.default, { className: "icon-close" }) })
            );
        }
    }]);
    return Frame;
}(_react2.default.Component);

exports.default = Frame;
module.exports = exports["default"];
